python/ray/data/tests/test_dataset.py

```diff
@@ -3566,6 +3566,40 @@ def get_node_id():
     assert set(locations) == {node1_id, node2_id}
 
 
+def test_random_sample(ray_start_regular_shared):
```

Comment:
Add a test as mentioned above - 5 blocks each has 1 row and we want to sample 20% ?

Reply:
Added tests in c9c6eb3